src/nemos/basis/_basis_mixin.py

```diff
@@ -0,0 +1,260 @@
+"""Mixin classes for basis."""
+
+from __future__ import annotations
+
+import copy
+import inspect
+from typing import Optional, Tuple, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+from ..convolve import create_convolutional_predictor
+from ._transformer_basis import TransformerBasis
+
+
+class EvalBasisMixin:
+    """Mixin class for evaluational basis."""
+
+    def __init__(self, bounds: Optional[Tuple[float, float]] = None):
+        self.bounds = bounds
+
+    def _compute_features(self, *xi: ArrayLike):
+        """
+        Apply the basis transformation to the input data.
+
+        The basis evaluated at the samples, or :math:`b_i(*xi)`, where :math:`b_i` is a
+        basis element. xi[k] must be a one-dimensional array or a pynapple Tsd.
```

Comment:
fixed the typing, also this are all edited in the PR in which i enable multi-dimensional inputs so we should  disregards docstrings (I already edited this one but it will create a conflict)